libs/base/Data/SnocList.idr

```diff
@@ -21,6 +21,14 @@ public export
 asList : SnocList type -> List type
 asList = (reverse . cast)
 
+export
+Uninhabited (Lin = x :< xs) where
+  uninhabited Refl impossible
+
+export
+Uninhabited (x :< xs = Lin) where
+  uninhabited Refl impossible
```

Comment:
i've made a rough guess where these and `Biinjective` should go

Reply:
It probably in a good place, but you should include `Data.SnocList` in `Decidable.Equality` then